core/tests/protractor/learnerDashboard.js

```diff
@@ -99,88 +97,90 @@ describe('Learner dashboard functionality', function() {
     workflow.publishExploration();
   };
 
-  it('displays completed explorations', function() {
+  it('displays incomplete and completed explorations', function() {
     users.createUser('originalCreator@learnerDashboard.com',
       'originalCreator');
     users.createUser('learner@learnerDashboard.com',
       'learnerlearnerDashboard');
     users.createAdmin('inspector@learnerDashboard.com', 'inspector');
     users.login('originalCreator@learnerDashboard.com');
-    // Create first exploration named 'About Oppia'
+    // Create exploration 'About Oppia'
     createAboutOppiaExploration();
     // Create a second exploration named 'Dummy Exploration'.
-    workflow.createAndPublishExploration(
-      'Dummy Exploration',
-      'Algebra',
-      'To learn about Algebra!',
-      'English'
-    );
     users.logout();
+
     users.login('learner@learnerDashboard.com');
-    // TODO(hoangviet1993): Leave the exploration in between.
-    // The exploration should be found in the 'In Progress' section.
-    // Expecting Alert window after leaving and unable to handle the alert at
-    // the point of writing this.
-    // Issue actively discussed here:
-    // https://github.com/angular/protractor/issues/308
-    // Play an exploration and leave it in between.
-
-    // Play exploration completely.
+    // Play exploration 'About Oppia'.
     libraryPage.get();
     libraryPage.findExploration('About Oppia');
     libraryPage.playExploration('About Oppia');
+    explorationPlayerPage.expectExplorationNameToBe('About Oppia');
     explorationPlayerPage.submitAnswer('Continue', null);
-    explorationPlayerPage.submitAnswer(
-      'MultipleChoiceInput', 'Those were all the questions I had!');
+    explorationPlayerPage.expectExplorationToNotBeOver();
+
+    // Disable waiting for Angular to accept alert.
```

Comment:
Maybe elaborate on this comment a bit

Can we abstract this chunk into a general library function? Seems a bit low-level for inclusion in the main test.

Reply:
OK. I have been looking at the [source code](https://github.com/angular/protractor/blob/a04435916101e8cf367333f3e7a51d41964e83ef/lib/browser.ts#L420) for .waitForAngularEnabled().

I do not feel comfortable explaining what it does in my own terms. Can you help me with abstracting it away too? Should I name it `general.safePageRefresh()` ? 

Details: Leaving mid-play causes an alert window which cannot be handled unless I disabled waitForAngular(). Once it is disabled, I accept the alert window and re-enable waitForAngular() and proceed as normal.

Reply:
Also, I do feel a tinge of uneasiness after adding `waitForAngularEnabled` in ... one of those " proceed with caution "  coming from my minimal knowledge about it. On [SO](https://stackoverflow.com/questions/42648077/how-does-waitforangularenabled-work/44073142) , it is said to isolate tests that have it to prevent flakiness. What do you think?

Reply:
Just say exactly what you've said in the last paragraph: "Leaving mid-play results in an alert window which cannot be handled unless waitForAngular() is disabled. Once it is disabled, we accept the alert, re-enable waitForAngular(), and proceed as normal."

In terms of abstracting -- safelyAcceptAlert() perhaps?

Reply:
Re second comment -- don't know, I don't have enough experience with this. My suggestion would be to keep it and file an issue to deal with it later, consistent with the aim of getting this PR in soonish.

Reply:
I named it safeAcceptAlert since acceptAlert already exists and share some of the codes.

Also, I have pretty gnarly result with learnerDashboard 's consistency once I put in the safeAcceptAlert() changes. I am trying to verify if learnerDashboard does better without it. I just really do not want it to be the next "editorFeatures".